tests/drivers/candev/main.c

```diff
@@ -19,42 +19,38 @@
  * @}
  */
 
-#include <assert.h>
-#include <errno.h>
 #include <isrpipe.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include "shell.h"
-#include "test_utils/expect.h"
-#include "can/device.h"
-
-#if IS_USED(MODULE_PERIPH_CAN)
 
-#include "periph/can.h"
-#include "can_params.h"
 #include "board.h"
-#include "periph_conf.h"
-#include "periph/gpio.h"
-
-static can_t periph_dev;
-
-#elif defined(MODULE_MCP2515)
+#include "can_params.h"
 #include "candev_mcp2515.h"
-#include "mcp2515_params.h"
+#include "periph/can.h"
+#include "periph_conf.h"
+#include "shell.h"
+#include "test_utils/expect.h"
 
-static candev_mcp2515_t mcp2515_dev;
+/* The params header is only in the include path when the module is used */
+#if MODULE_MCP2515
+#  include "mcp2515_params.h"
+#endif
 
-#else
-/* add includes for other candev drivers here */
+#ifdef BOARD_SAME54_XPRO
+#  include "periph/gpio.h"
```

Comment:
Why only for this board?

Reply:
That is annoying, isn't it?

I'll add a `__attribute__((weak)) board_enable_can_transceiver(can_t *dev);` and a `__attribute__((weak)) board_disable_can_transceiver(can_t *dev);` into the `periph_can` driver, so that we do not need to have magic stuff at application level, but rather let the board hook into `can_init()`.

But that is not independent of this PR and will be a follow up.

Reply:
We already have a [CAN transceiver interface](https://doc.riot-os.org/group__drivers__can__trx.html), it's just not integrated with candev :weary: